datamonster_api/lib/datasource.py

```diff
@@ -4,26 +4,39 @@
 
 
 class Datasource(BaseClass):
-    """Class for a datasource"""
+    """Datasource object which represents a data source in DataMonster
 
-    def __init__(self, id_, name, category, uri, dm):
-        self.id = id_
-        self.name = name
-        self.category = category
-        self.uri = uri
-        self.dm = dm
+    :param params: (dict)
+    :param dm: DataMonster object
+
+    Attributes include:
+      name: (str) datasource name
+      category: (str) category associated with the datasource
+    """
+
+    def __init__(self, params, dm):
+        self._dm = dm
+        self._params = params
+        self._id = params["id"]
+        self._uri = params["uri"]
+        self.name = params["name"]
+        self.category = params["category"]
 
     def get_details(self):
         """Get details (metadata) for this datasource
+
+        :return: (dict) details
         """
-        return self.dm.get_datasource_details(self.id)
+        return self._dm.get_datasource_details(self._id)
 
     @property
     def companies(self):
-        """Return the (memoized) companies for this data source.
+        """get the (memoized) companies for this data source.
+
+        :return: (iter) iterable of Company objects
         """
         if not hasattr(self, "_companies"):
```

Comment:
Yeah, now that I see us doing more than one of these, I think we should add memoized_property to clean it up

Reply:
updated - nice!

Reply:
Yeah, it's a nice little function for exactly that :-)